sdk/identity/azure-identity/azure/identity/_credentials/cli_credential.py

```diff
@@ -0,0 +1,83 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import sys
+_IS_PY2 = sys.version_info[0] < 3
+
+import platform
+_PLATFORM_STR = platform.system()
+_PLATFORM = 1 if _PLATFORM_STR == 'Windows' else 2 if _PLATFORM_STR == 'Linux' else 3
+_USE_SHELL = True if _PLATFORM == 1 else False
+
+import json
+import time
+from datetime import datetime
+
+from subprocess import check_output, check_call, CalledProcessError, STDOUT, PIPE
+try:
+    from subprocess import run
+except:
+    pass
+
+from azure.core.credentials import AccessToken
+from azure.core.exceptions import ClientAuthenticationError
+
+class CliCredential(object):
+
+    _LINUX_CMD_NOT_FOUND = 'command not found'
+    _WINDOWS_CMD_NOT_FOUND = 'is not recognized as'
+    _DEFAULT_PREFIX = "/.default"
+    _CLI_NOT_INSTALLED_ERR = "Azure CLI not installed"
+
+    def get_token(self, *scopes, **kwargs): # pylint:disable=unused-argument
+        command = ['az', 'account', 'get-access-token']
+
+        if scopes:
+            resource = scopes[0]
+            if resource.endswith(self._DEFAULT_PREFIX):
+                resource = resource[:-len(self._DEFAULT_PREFIX)]
+
+            command.extend(['--resource', resource])
+
+        if _IS_PY2:
```

Comment:
What is the value to implement it twice? Why not just keeping the Py2 version for everything?

Reply:
Okay, I'll go check_output() for both Python 2 and 3. But still, this method has different behaviors in the two versions.